RecoHGCal/TICL/python/ticl_iterations.py

```diff
@@ -10,109 +10,153 @@
 from RecoLocalCalo.HGCalRecProducers.hgcalLayerClusters_cff import hgcalLayerClusters
 from RecoLocalCalo.HGCalRecProducers.hgcalMultiClusters_cfi import hgcalMultiClusters
 
-
+from RecoHGCal.TICL.ticlSeedingRegionProducer_cfi import ticlSeedingRegionProducer
 from RecoHGCal.TICL.ticlLayerTileProducer_cfi import ticlLayerTileProducer
 from RecoHGCal.TICL.trackstersProducer_cfi import trackstersProducer
 from RecoHGCal.TICL.filteredLayerClustersProducer_cfi import filteredLayerClustersProducer
 from RecoHGCal.TICL.multiClustersFromTrackstersProducer_cfi import multiClustersFromTrackstersProducer
 
-
+## withReco: requires full reco of the event to run this part
+## i.e. collections of generalTracks can be accessed
 def TICL_iterations_withReco(process):
-  process.FEVTDEBUGHLTEventContent.outputCommands.extend(['keep *_MultiClustersFromTracksters*_*_*'])
+  process.FEVTDEBUGHLTEventContent.outputCommands.extend(['keep *_multiClustersFromTracksters*_*_*'])
+
+  process.ticlLayerTileProducer = ticlLayerTileProducer.clone()
+
+  process.ticlSeedingTrk = ticlSeedingRegionProducer.clone(
+    algoId = 1
+  )
+
+  process.filteredLayerClustersTrk = filteredLayerClustersProducer.clone(
+    clusterFilter = "ClusterFilterByAlgo",
+    algo_number = 8,
+    iteration_label = "Trk"
+  )
 
-  process.TICLLayerTileProducer = ticlLayerTileProducer.clone()
+  process.trackstersTrk = trackstersProducer.clone(
+    filtered_mask = cms.InputTag("filteredLayerClustersTrk", "Trk"),
+    seeding_regions = cms.InputTag("ticlSeedingTrk"),
```

Comment:
please drop type specifications for all parameters which already exist.
This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

Reply:
> please drop type specifications for all parameters which already exist.
> This is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.

this was not addressed yet

Reply:
Sorry for not commenting before, I don't understand what should be dropped here.
Removing "seeding_regions = cms.InputTag("ticlSeedingTrk")," produces the following exception...
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<TICLSeedingRegion>
Looking for module label: ticlSeedingRegionProducer
Looking for productInstanceName:

Reply:
```cms.InputTag``` is a type specification, I was asking to drop it. => ```seeding_regions = "ticlSeedingTrk"```

Reply:
ok, thanks for the clarification